src/scaraplate/strategies.py

```diff
@@ -320,19 +340,24 @@ def _maybe_add_linter_ignore(self, line: str) -> str:
         return line
 
     def apply(self) -> BinaryIO:
-        searched_comment = self.render_comment(self.searched_comment_contents()).encode(
-            "ascii"
+        newline = detect_newline(self.target_contents, self.template_contents)
+
+        searched_comment = self.render_comment(
+            self.searched_comment_contents(), newline=newline
         )
-        appended_comment = self.render_comment(self.comment_contents()).encode("ascii")
+        appended_comment = self.render_comment(self.comment_contents(), newline=newline)
         if self.target_contents is not None:
             target_text = self.target_contents.read()
             if searched_comment in target_text and not self.template_meta.is_git_dirty:
                 # Hash hasn't changed -- keep the target.
                 self.target_contents.seek(0)
                 return self.target_contents
 
-        out_bytes = self.template_contents.read()
-        out_bytes += b"\n" + appended_comment  # TODO detect newlines type?
+        # Convert template newlines to the target file newlines:
+        template_lines = self.template_contents.read().splitlines()
+        out_bytes = b"".join(line + newline for line in template_lines)
```

Comment:
`newline.join(template_lines)` ?

Reply:
That would have skipped a trailing newline.

Reply:
Added one more test which ensures that the trailing newline is kept in both cases: when the template file ends with one and when don't. https://github.com/rambler-digital-solutions/scaraplate/blob/b0a5f555e79210e11512bb20dc32d517fd5a755c/tests/strategies/test_template_hash.yml#L95-L115

`splitlines` provides identical output in both cases, so that `join` effectively would ensure that the template file ends with a newline (which, I believe, is a good thing: having an empty line between the template contents and the appended comment helps to visually distinguish between the two).

Reply:
Good!